src/core/lib/support/thd_posix.cc

```diff
@@ -33,17 +33,31 @@
 struct thd_arg {
   void (*body)(void* arg); /* body of a thread */
   void* arg;               /* argument to a thread */
+  const char* name;        /* name of thread */
 };
 
 /* Body of every thread started via gpr_thd_new. */
 static void* thread_body(void* v) {
   struct thd_arg a = *(struct thd_arg*)v;
   free(v);
+  if (a.name != NULL) {
+#if GPR_APPLE_PTHREAD_NAME
+    /* Apple supports 64 characters, and will truncate if it's longer. */
+    pthread_setname_np(a.name);
+#elif GPR_LINUX_PTHREAD_NAME
+    /* Linux supports 16 characters max, and will error if it's longer. */
+    char buf[16];
+    strncpy(buf, a.name, sizeof(buf) - 1);
```

Comment:
you can use `GPR_ARRAY_SIZE` for `buf` here.

Reply:
Done